src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java

```diff
@@ -0,0 +1,187 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check ensures that roundabouts with unreasonable valences are flagged. In reference to OSM
+ * Wiki, each roundabout should have greater than 1 connection as 1 connection should be tagged as a
+ * turning point, and no connections is obviously not a valid way.
+ *
+ * @author savannahostrowski
+ */
+public class RoundaboutValenceCheck extends BaseCheck
+{
+
+    private static final long serialVersionUID = 1L;
+    public static final String WRONG_VALENCE_INSTRUCTIONS = "This roundabout, {0, number, #}, "
+            + "has the wrong valence. It has a valence of {1, number, integer}.";
+    public static final String VALENCE_OF_ONE_INSTRUCTIONS = "This feature, {0,number,#},"
+            + " should not be labelled as a roundabout. "
+            + "This feature should be a turning loop or turning circle.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(WRONG_VALENCE_INSTRUCTIONS, VALENCE_OF_ONE_INSTRUCTIONS);
+
+    private static final double LOWER_VALENCE_THRESHOLD_DEFAULT = 2.0;
+    private static final double UPPER_VALENCE_THRESHOLD_DEFAULT = 10.0;
```

Comment:
Can you update this to match the description? Or is this as intended?

Reply:
Can you clarify? @jklamer

Reply:
In the description it says "By default, unusual valence is defined as less than 2 or greater than 14". And I just wanted to make sure the ```UPPER_VALENCE_THRESHOLD_DEFAULT = 10``` is intended.

Reply:
Oh yeah, that's intended.